### PR TITLE
glue: Implement the connection infrastructure between xsim and C++

### PR DESCRIPTION
**Pull request by ljanyst**
This PR implements the build and the glue infrastructure between xsim and C++.

### CODE REVIEW

**Review by Richard-Gordon — APPROVED** [review]
This meets my expectations and I see no errors. I suggest a more experienced software engineer also reviews this; the interaction with sockets and message sizing is not intuitive to me, for example.

File: src/common/Utils.cc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    buffer.resize(size);
+    uint8_t *ptr = buffer.data();
+
+    while (size) {
+        ssize_t rd = read(sock, ptr, size);
+        if (rd == -1) {
+            return -1;
+        }
+        size -= rd;
```
Comment (jfrohnhofen) [inline src/common/Utils.cc:46]: What happens if you get more data than expected? In that case size will underflow and you'll keep reading forever.

Comment (jfrohnhofen) [reply src/common/Utils.cc:46]: Not sure if this case can happen and if this is a concern.

File: src/sim/bridge.cc [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+/**
+ * Bridge is the SystemVerilog DPI counterpart of the `sw_axi::Bridge`.
+ *
+ * It implements the communication mechanism with the software side of the system.
+ */
+class Bridge {
+public:
+    /**
+     * Create a server socket at the given randez-vous point and wait for the client t connect
```
Comment (jfrohnhofen) [inline src/sim/bridge.cc]: t -> to

File: src/sim/bridge.cc [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    char buff;
+    read(sock, &buff, 1);
+    disconnect();
+
+    LOG << "[SW-AXI] Client disconnected from: " << connectedUri << std::endl;
+}
+}  // namespace
+
+extern "C" void *sw_axi_connect(const char *uri) {
+    Bridge *bridge = new Bridge;
```
Comment (jfrohnhofen) [inline src/sim/bridge.cc:191]: This is never freed, obviously. But maybe okay for the current use-case. Or this could be freed in sw_axi_disconnect

**Review by jfrohnhofen — APPROVED** [review]
Looking good. Added a few minor comments.

File: src/common/Utils.cc (hunk 1 above)
Comment (ljanyst) [reply src/common/Utils.cc:46]: You declare as a parameter to `read` the max you're willing to take. You will not get more. You may get less, though, depending on the state of the sockets, hence the loop.

File: src/sim/bridge.cc (hunk 2 above)
Comment (ljanyst) [reply src/sim/bridge.cc]: Merci!

File: src/sim/bridge.cc (hunk 3 above)
Comment (ljanyst) [reply src/sim/bridge.cc:191]: Yup. Will fix. Thanks!

**Pull request merged**